crates/eips/src/eip2930.rs

```diff
@@ -130,15 +130,17 @@ impl AccessList {
     }
 }
 
-/// Access list with gas used appended.
+/// `AccessListResult` for handling errors from `eth_createAccessList`
 #[derive(Clone, Debug, Default, PartialEq, Eq)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
-#[cfg_attr(feature = "serde", serde(rename_all = "camelCase"))]
-pub struct AccessListWithGasUsed {
```

Comment:
I'd like to keep this type as well

so we can add another function that is

```rust
impl AccessListResult {

  pub fn ensure_ok(self) -> Result<AccessListWithGasUsed, String>

} 

```

and `fn is_err(&self) {self.error.is_some()}`